spec/display-buffer-spec.coffee

```diff
@@ -147,19 +144,13 @@ describe "DisplayBuffer", ->
         it "includes hanging indent when breaking soft-wrap indentation into tokens", ->
           tokenizedLine = displayBuffer.tokenizedLineForScreenRow(4)
 
-          expect(tokenizedLine.tokens[0].value).toBe("  ")
+          expect(tokenizedLine.tokens[0].value).toBe("    ")
           expect(tokenizedLine.tokens[0].isSoftWrapIndentation).toBeTruthy()
 
-          expect(tokenizedLine.tokens[1].value).toBe("  ")
+          expect(tokenizedLine.tokens[1].value).toBe("   ") # hanging indent
           expect(tokenizedLine.tokens[1].isSoftWrapIndentation).toBeTruthy()
 
-          expect(tokenizedLine.tokens[2].value).toBe("  ") # hanging indent
```

Comment:
We actually want to preserve these tests. What about setting `tabLength` only on `::clipScreenPosition` test and leave everything as-is? Here's a diff you can apply to your current branch:

```diff
diff --git a/spec/display-buffer-spec.coffee b/spec/display-buffer-spec.coffee
index 9b38323..361e48c 100644
--- a/spec/display-buffer-spec.coffee
+++ b/spec/display-buffer-spec.coffee
@@ -4,7 +4,7 @@ _ = require 'underscore-plus'
 describe "DisplayBuffer", ->
   [displayBuffer, buffer, changeHandler, tabLength] = []
   beforeEach ->
-    tabLength = 4
+    tabLength = 2
 
     buffer = atom.project.bufferForPathSync('sample.js')
     displayBuffer = new DisplayBuffer({buffer, tabLength})
@@ -127,10 +127,13 @@ describe "DisplayBuffer", ->
         it "breaks soft-wrap indentation into a token for each indentation level to support indent guides", ->
           tokenizedLine = displayBuffer.tokenizedLineForScreenRow(4)
 
-          expect(tokenizedLine.tokens[0].value).toBe("    ")
+          expect(tokenizedLine.tokens[0].value).toBe("  ")
           expect(tokenizedLine.tokens[0].isSoftWrapIndentation).toBeTruthy()
 
-          expect(tokenizedLine.tokens[1].isSoftWrapIndentation).toBeFalsy()
+          expect(tokenizedLine.tokens[1].value).toBe("  ")
+          expect(tokenizedLine.tokens[1].isSoftWrapIndentation).toBeTruthy()
+
+          expect(tokenizedLine.tokens[2].isSoftWrapIndentation).toBeFalsy()
 
       describe "when editor.softWrapHangingIndent is set", ->
         beforeEach ->
@@ -144,13 +147,19 @@ describe "DisplayBuffer", ->
         it "includes hanging indent when breaking soft-wrap indentation into tokens", ->
           tokenizedLine = displayBuffer.tokenizedLineForScreenRow(4)
 
-          expect(tokenizedLine.tokens[0].value).toBe("    ")
+          expect(tokenizedLine.tokens[0].value).toBe("  ")
           expect(tokenizedLine.tokens[0].isSoftWrapIndentation).toBeTruthy()
 
-          expect(tokenizedLine.tokens[1].value).toBe("   ") # hanging indent
+          expect(tokenizedLine.tokens[1].value).toBe("  ")
           expect(tokenizedLine.tokens[1].isSoftWrapIndentation).toBeTruthy()
 
-          expect(tokenizedLine.tokens[2].isSoftWrapIndentation).toBeFalsy()
+          expect(tokenizedLine.tokens[2].value).toBe("  ") # hanging indent
+          expect(tokenizedLine.tokens[2].isSoftWrapIndentation).toBeTruthy()
+
+          expect(tokenizedLine.tokens[3].value).toBe(" ") # odd space
+          expect(tokenizedLine.tokens[3].isSoftWrapIndentation).toBeTruthy()
+
+          expect(tokenizedLine.tokens[4].isSoftWrapIndentation).toBeFalsy()
 
     describe "when the buffer changes", ->
       describe "when buffer lines are updated", ->
@@ -615,6 +624,9 @@ describe "DisplayBuffer", ->
 
   describe "::clipScreenPosition(screenPosition, wrapBeyondNewlines: false, wrapAtSoftNewlines: false, clipToAtomicTokenEdge: 'closest')", ->
     beforeEach ->
+      tabLength = 4
+
+      displayBuffer.setTabLength(tabLength)
       displayBuffer.setSoftWrapped(true)
       displayBuffer.setEditorWidthInChars(50)
 
@@ -716,7 +728,6 @@ describe "DisplayBuffer", ->
 
     it "correctly translates positions on soft wrapped lines containing tabs", ->
       buffer.setText('\t\taa  bb  cc  dd  ee  ff  gg')
-      displayBuffer.setTabLength(2)
       displayBuffer.setSoftWrapped(true)
       displayBuffer.setEditorWidthInChars(10)
       expect(displayBuffer.screenPositionForBufferPosition([0, 10], wrapAtSoftNewlines: true)).toEqual [1, 4]

```